src/github.ts

```diff
@@ -41,26 +56,57 @@ export function getTempDir(): string {
 
 function getPullRequestContext(
   context: Context
-): Pick<GitHubContext, "prNumber" | "sha"> {
-  const number = context.issue.number;
+): PullRequestInfo {
+  const prNumber = context.issue.number;
   const sha = context.payload.pull_request?.head.sha;
-  return { prNumber: number, sha };
+
+  return { prNumber, sha };
 }
 
 function getCheckRunContext(
   context: Context
-): Pick<GitHubContext, "prNumber" | "sha"> {
+): PullRequestInfo {
   const actionEvent = context.payload.check_run;
-  const number = actionEvent.pull_requests?.[0]?.number;
+  const prNumber = actionEvent.pull_requests?.[0]?.number;
   const sha = actionEvent.head_sha;
-  return { prNumber: number, sha };
+
+  return { prNumber, sha };
+}
+
+/**
+ * Retrieves commit information for a pull request from the GitHub API if the workflow
+ * is triggered from a specified branch. Otherwise, returns the commit information
+ * for the current branch.
+ * @param context The GitHub context object containing information about the event.
+ * @returns A Promise that resolves to a CommitInfo object with the SHA of the commit.
+ */
+async function getWorkflowDispatchContext(context: Context): Promise<PullRequestInfo> {
+  const branchName: string = context.ref.substring('refs/heads/'.length);
+  const allowedBranches: string[] = ['main', 'develop'];
```

Comment:
Rename this to defaultBranches. 

BUT it will be better that we dont have to keep this list and isntead get the name of the default branch something. I found this tha could help you: https://stackoverflow.com/questions/64781462/github-actions-default-branch-variable